datashader/transfer_functions/__init__.py

```diff
@@ -172,20 +172,22 @@ def eq_hist(data, mask=None, nbins=256*256):
 
     # Run more accurate value counting if data is of boolean or integer type
     # and unique value array is smaller than nbins.
-    if data2.dtype == bool or (np.issubdtype(data2.dtype, np.integer) and data2.max() < nbins):
+    if data2.dtype == bool or (np.issubdtype(data2.dtype, np.integer) and data2.ptp() < nbins):
```

Comment:
Makes sense, though there's always a non-zero minimum because zero values are treated as NaN, so really this better covers the case when there is a minimum other than 1.